provider/pkg/provider/helm_release_test.go

```diff
@@ -138,3 +140,41 @@ func Test_MergeMaps(t *testing.T) {
 		})
 	}
 }
+
+func TestDecodeRelease(t *testing.T) {
+	tests := []struct {
+		name  string
+		given resource.PropertyMap
+		want  *Release
+	}{
+		{
+			name: "valueYamlFiles layering",
+			given: resource.PropertyMap{
+				"valueYamlFiles": resource.NewArrayProperty([]resource.PropertyValue{
```

Comment:
Maybe this test should also include a literal value, to test how the value files combine with literals.  e.g. a literal value of `repository: bitnami/nginx-patched`.

This test should check the relative priority of the files, by setting `tag: latest` into the first file (and would be superseded by the second file).